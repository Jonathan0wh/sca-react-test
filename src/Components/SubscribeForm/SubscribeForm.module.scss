@import "styles/colors";

.form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.form_container {
  flex-direction: column;
  justify-content: center;
  align-items: center;
  @media (max-width: 768px) {
    width: 90%;
  }
}

.succsss_wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  color: $white;
  font-size: larger;
  font-weight: bold;
  height: 75%;
}
