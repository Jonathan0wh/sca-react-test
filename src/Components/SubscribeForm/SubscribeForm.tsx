import React from 'react';
import classNames from 'classnames';

import styles from './SubscribeForm.module.scss';
import Input from 'Components/Shared/Input';
import Button from 'Components/Shared/Button';

const SubscribeForm = () => (
  <form action="" method="post" className={styles.form}>
    <div className={classNames('container', styles.form_container)}>
      <div className="row">
        <div className="col-md-6">
          <Input labelName="First Name" inputType="text" required />
        </div>
        <div className="col-md-6">
          <Input labelName="Last Name" inputType="text" required />
        </div>
      </div>

      <div className="row">
        <div className="col-md-6">
          <Input labelName="Email" inputType="email" required />
        </div>
        <div className="col-md-6">
          <Input labelName="Mobile Phone" inputType="number" required={false} />
        </div>
      </div>

      <div className="row">
        <Button buttonText="Submit" />
      </div>
    </div>
  </form>
);

export default SubscribeForm;
