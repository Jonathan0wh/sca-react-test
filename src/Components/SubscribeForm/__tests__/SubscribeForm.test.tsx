import React from 'react';
import { shallow } from 'enzyme';

import SubscribeForm from '../SubscribeForm';

describe('<SubscribeForm />', () => {
  it('render form by default', () => {
    const wrapper = shallow(<SubscribeForm />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render form with some text entered', () => {
    const wrapper = shallow(<SubscribeForm />);
    wrapper.setState({
      firstName: 'test First Name',
      lastName: 'test Last Name',
      email: 'test@email.com',
      mobilePhone: '2121121',
      submitting: false,
      subscribeSuccess: false
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('render form when submitting', () => {
    const wrapper = shallow(<SubscribeForm />);
    wrapper.setState({
      firstName: 'test First Name',
      lastName: 'test Last Name',
      email: 'test@email.com',
      mobilePhone: '2121121',
      submitting: true,
      subscribeSuccess: false
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('render success message after submitting success', () => {
    const wrapper = shallow(<SubscribeForm />);
    wrapper.setState({
      firstName: 'test First Name',
      lastName: 'test Last Name',
      email: 'test@email.com',
      mobilePhone: '2121121',
      submitting: false,
      subscribeSuccess: true
    });

    expect(wrapper).toMatchSnapshot();
  });
});
