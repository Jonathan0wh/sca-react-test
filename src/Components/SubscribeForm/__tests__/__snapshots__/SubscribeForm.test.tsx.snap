// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscribeForm /> render form by default 1`] = `
<form
  className="form"
>
  <div
    className="container form_container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <Input
          dataField="firstName"
          handleInputChange={[Function]}
          inputType="text"
          labelName="First Name"
          required={true}
        />
      </div>
      <div
        className="col-md-6"
      >
        <Input
          dataField="lastName"
          handleInputChange={[Function]}
          inputType="text"
          labelName="Last Name"
          required={true}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <Input
          dataField="email"
          handleInputChange={[Function]}
          inputType="email"
          labelName="Email"
          required={true}
        />
      </div>
      <div
        className="col-md-6"
      >
        <Input
          dataField="mobilePhone"
          handleInputChange={[Function]}
          inputType="number"
          labelName="Mobile Phone"
          required={false}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <Button
        buttonText="Submit"
        handleClick={[Function]}
        loading={false}
      />
    </div>
  </div>
</form>
`;

exports[`<SubscribeForm /> render form when submitting 1`] = `
<form
  className="form"
>
  <div
    className="container form_container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <Input
          dataField="firstName"
          handleInputChange={[Function]}
          inputType="text"
          labelName="First Name"
          required={true}
        />
      </div>
      <div
        className="col-md-6"
      >
        <Input
          dataField="lastName"
          handleInputChange={[Function]}
          inputType="text"
          labelName="Last Name"
          required={true}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <Input
          dataField="email"
          handleInputChange={[Function]}
          inputType="email"
          labelName="Email"
          required={true}
        />
      </div>
      <div
        className="col-md-6"
      >
        <Input
          dataField="mobilePhone"
          handleInputChange={[Function]}
          inputType="number"
          labelName="Mobile Phone"
          required={false}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <Button
        buttonText="Submit"
        handleClick={[Function]}
        loading={true}
      />
    </div>
  </div>
</form>
`;

exports[`<SubscribeForm /> render form with some text entered 1`] = `
<form
  className="form"
>
  <div
    className="container form_container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <Input
          dataField="firstName"
          handleInputChange={[Function]}
          inputType="text"
          labelName="First Name"
          required={true}
        />
      </div>
      <div
        className="col-md-6"
      >
        <Input
          dataField="lastName"
          handleInputChange={[Function]}
          inputType="text"
          labelName="Last Name"
          required={true}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <Input
          dataField="email"
          handleInputChange={[Function]}
          inputType="email"
          labelName="Email"
          required={true}
        />
      </div>
      <div
        className="col-md-6"
      >
        <Input
          dataField="mobilePhone"
          handleInputChange={[Function]}
          inputType="number"
          labelName="Mobile Phone"
          required={false}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <Button
        buttonText="Submit"
        handleClick={[Function]}
        loading={false}
      />
    </div>
  </div>
</form>
`;

exports[`<SubscribeForm /> render success message after submitting success 1`] = `
<div
  className="succsss_wrapper"
>
  Thank you for subscribing!
</div>
`;
